snapshot/restore/local_fs_output.go

```diff
@@ -326,6 +336,81 @@ func write(targetPath string, r fs.Reader) error {
 	return nil
 }
 
+type sparseWriter struct {
+	f *os.File
+}
+
+// nolint:wrapcheck
+func (sw *sparseWriter) Write(b []byte) (int, error) {
+	w := 0
+
+	bSize, err := stat.GetFileAllocSize(sw.f.Name())
```

Comment:
this will be invoked tons of times (every 64KB), perhaps do it outside and pass as a field?

Reply:
Ah right, thanks for pointing that out! We can make that part of the `sparseWriter` struct.